### PR TITLE
Install script

### PR DESCRIPTION
**Pull request by xuqingfeng**
- `curl -fsSL https://raw.githubusercontent.com/Wiredcraft/pipelines/install-script/install.sh | sh -s dev/stable`

- `./install.sh dev/stable`

@zbal / @JuhaS  got a minute to review?

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    if [ ! -n "$(which pip)" ]; then
+        echo "Need pip installed !";
+        exit 1;
+    fi
+    if [ ! -n "$(which git)" ]; then
+        echo "Need git installed !"
+        exit 1;
+    fi
+
+    echo "Installing pipelines..."
```
Comment (zbal) [inline install.sh]: I'd add a step to ask whether the user wants the latest dev or the mainstream version

```
echo -n "Which version of pipelines to install? (dev/LATEST) "
read version
if [ -z "$version" ]; then
    version=LATEST
fi

if [ "$version" == "dev" ]; then
    sudo pip install git+git://github.com/Wiredcraft/pipelines@dev > /dev/null
elif [ "$version" == "LATEST" -o "$version" == "latest" ]; then
    sudo pip install pipelines
else
    echo "Invalid version."
    exit 1
fi
```

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    sudo pip install git+git://github.com/Wiredcraft/pipelines@dev > /dev/null
+
+    echo "Creating user..."
+    id pipelines > /dev/null 2>&1
+    if [ ! $? -eq 0 ]; then
```
Comment (zbal) [inline install.sh]: :)

```
if [ $? -ne 0 ]; then
```

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        sudo useradd -m -d /var/lib/pipelines -s /bin/false pipelines
+    fi
+
+    echo "Creating workspace..."
+    sudo mkdir -p /var/lib/pipelines/workspace
+    sudo chown -R pipelines:pipelines /var/lib/pipelines
+
```
Comment (zbal) [inline install.sh]: Add SSH creation

```
sudo -u pipelines ssh-keygen -P '' -f /var/lib/pipelines/.ssh/id_rsa -b 2048 -q
echo "Pipelines' SSH public KEY:"
echo
cat /var/lib/pipelines/.ssh/id_rsa.pub
echo
```

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    echo ""
+    pipelines --version
+    echo ""
+
+    echo "Successfully installed"
```
Comment (zbal) [inline install.sh]: Can we also add `supervisor` setup?

```
echo "We recommend to run pipelines via supervisor"
echo -n "Proceed with the setup? [Y/n] " 
read install_supervisor

if [ -z "$install_supervisor" -o "$install_supervisor" == "y" ]; then
  install_supervisor
fi

install_supervisor() {
if [ REDHAT ]; then
    yum install -y epel-release
    yum install -y supervisord
elif [ DEBIAN ]; then
    apt-get install supervisor
else;
    echo "Neither Debian nor RH based system - install supervisor manually, or other process manager"
fi

echo "Adding supervisor config file for pipelines"
# Path may be RH / Debian specific
cp xyz /etc/supervisor/conf.d/pipelines.conf
}
```

File: install.sh (hunk 1 above)
Comment (xuqingfeng) [inline install.sh]: I think `stable` is better.

File: install.sh (hunk 4 above)
Comment (xuqingfeng) [inline install.sh]: Not now 😂

File: install.sh [hunk 5]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
+
+set -e
+
+install_pipelines(){
+
```
Comment (zbal) [inline install.sh]:
```
if [ -z "$1" ]; then
    help
    exit 
fi 
```

**Comment by zbal** [comment]
as soon as the "help" is added @sunsongxp - please go ahead and merge.

**Comment by xuqingfeng** [comment]
@sunsongxp done

**Comment by zbal** [comment]
Let's get moving - not optimal - can fix later...

**Pull request merged**